crates/forge_services/src/preferences.rs

```diff
@@ -59,6 +70,13 @@ impl<F: ProviderRepository + AppConfigRepository + Send + Sync> AppConfigService
         &self,
         provider_id: Option<&ProviderId>,
     ) -> anyhow::Result<ModelId> {
+        use forge_domain::ModelId;
+
+        // Check for environment variable override first
+        if let Some(model_str) = self.infra.get_env_var("FORGE_OVERRIDE_MODEL") {
+            return Ok(ModelId::new(&model_str));
+        }
+
```

Comment:
Why should `set` be affected?

Reply:
Its in get only please expand few lines above